docs/search.md

```diff
@@ -0,0 +1,139 @@
+
+* `on`: may be used with `eq` or `range` to limit results to a particular card type.
+
+Nested fields may be specified using a dot (`.`) notation.
+
+#### Examples
+
+```js
+  let { data: matching } = await indexer.search({
+    filter: {
+      type: { module: `https://my.realm/article`, name: 'Article' },
+    },
+  });
+```
+
+```js
+  let { data: matching } = await indexer.search({
+    filter: {
+      on: { module: `https://my.realm/post`, name: 'Post' },
+      range: { views: { lte: 10, gt: 5 }, 'author.posts': { gte: 1 } },
+    },
+  });
+```
+
+```js
+  let { data: matching } = await indexer.search({
+    filter: {
+      on: { module: `https://my.realm/booking`, name: 'Booking' },
+      eq: { 'hosts.firstName': 'Arthur' },
+    },
+  });
+```
+
+```js
+  let { data: matching } = await indexer.search({
+    filter: {
+      on: { module: `https://my.realm/article`, name: 'Article' },
+      not: { eq: { 'author.firstName': 'Carl' } },
+    },
+  });
+```
+
+```js
+  let { data: matching } = await indexer.search({
+    filter: {
+      any: [
+        {
+          on: { module: `https://my.realm/article`, name: 'Article' },
+          eq: { 'author.firstName': 'Kafka' },
+        },
+        {
+          on: { module: `https://my.realm/book`, name: 'Book' },
+          eq: { 'author.firstName': 'Kafka' },
+        },
+      ],
+    },
+  });
+```
+
+```js
+  let { data: matching } = await indexer.search({
+    filter: {
+      on: {
+        module: `https://my.realm/post`,
+        name: 'Post',
+      },
+      every: [
+        { eq: { title: 'Card 1' } },
+        { not: { eq: { 'author.firstName': 'Cardy' } } },
+      ],
+    },
+  });
+```
+
+### Sort Order
+
+The Query object may have a `sort` property, which controls the order in which results are returned.
+
+A `sort` is an array property value where each item is an object that consists of `on`, `by`, and optionally `direction`. The array is the order in which the sort is applied.
+
+* `on`: The card type being sorted
+* `by`: The field name to be sorted on. Nested fields may be supported using a dot (`.`) notation.
+* `direction`: `asc` of `desc`, default is `asc`
+
+#### Examples
+
+```js
+  let { data: matching } = await indexer.search({
+    sort: [
+      {
+        by: 'author.firstName',
+        on: { module: `https://my.realm/article`, name: 'Article' },
+        direction: 'desc',
+      },
+    ],
+    filter: { type: { module: `https://my.realm/post`, name: 'Post' } },
+  });
+```
+
+```js
+  let { data: matching } = await indexer.search({
+    sort: [
+      {
+        by: 'editions',
+        on: { module: `https://my.realm/book`, name: 'Book' },
+        direction: 'desc',
+      },
+      {
+        by: 'author.lastName',
+        on: { module: `https://my.realm/book`, name: 'Book' },
+      },
+    ],
+    filter: { type: { module: `https://my.realm/book`, name: 'Book' } },
+  });
+```
+
+### Pagination -- TODO: is this used yet????
```

Comment:
yeah this is still TBD. I'm guessing we wont really tackle this until we have a DB based index

Reply:
👍